python/pyiceberg/io/pyarrow.py

```diff
@@ -798,20 +800,16 @@ def _task_to_table(
             else:
                 arrow_table = fragment_scanner.to_table()
 
-        if limit:
-            with rows_counter:
-                if rows_counter.value >= limit:
-                    return None
-                rows_counter.value += len(arrow_table)
+        if len(arrow_table) < 1:
+            return None
 
-        # If there is no data, we don't have to go through the schema
-        if len(arrow_table) > 0:
-            return to_requested_schema(projected_schema, file_project_schema, arrow_table)
-        else:
+        if limit and row_count >= limit:
```

Comment:
If the limit would be 0, then it would just pass:
```python
Python 3.9.6 (default, May  7 2023, 23:32:44) 
[Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> if 0:
...     print('vo')
... 
>>> 
```


```suggestion
        if limit is not None and row_count >= limit:
```